hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -966,6 +967,126 @@ int SaraNcpClient::checkNetConfForImsi() {
+                    }
+                } else {
+                    newProf = static_cast<int>(UbloxSaraUmnoprof::STANDARD_EUROPE);
+                }
+            }
+            // KORE AT&T or 3rd Party SIM
+            else {
+                // break out of do-while if we're trying to set SIM_SELECT a third time
+                if (resetCount >= 2) {
+                    LOG(WARN, "UMNOPROF=1 did not resolve a built-in profile, please check if UMNOPROF=100 is required!");
+                    break;
+                }
+            }
+
+            // Disconnect before making changes to the UMNOPROF
+            auto respCfun = parser_.sendCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN?");
+            int cfunVal = -1;
+            auto retCfun = CHECK_PARSER(respCfun.scanf("+CFUN: %d", &cfunVal));
+            CHECK_PARSER_OK(respCfun.readResult());
+            if (retCfun == 1 && cfunVal != 0) {
+                CHECK_PARSER_OK(parser_.execCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN=0,0"));
+            }
+            // This is a persistent setting
+            parser_.execCommand(1000, "AT+UMNOPROF=%d", newProf);
+            // Not checking for error since we will reset either way
+            reset = true;
+            disableLowPowerModes = true;
+        } else if (r == 1 && static_cast<UbloxSaraUmnoprof>(curProf) == UbloxSaraUmnoprof::STANDARD_EUROPE) {
+            auto respBand = parser_.sendCommand(UBLOX_UBANDMASK_TIMEOUT, "AT+UBANDMASK?");
+            uint64_t ubandUint64 = 0;
+            char ubandStr[24] = {};
+            auto retBand = CHECK_PARSER(respBand.scanf("+UBANDMASK: 0,%23[^,]", ubandStr));
+            CHECK_PARSER_OK(respBand.readResult());
+            if (netConf_.netProv() == CellularNetworkProvider::TWILIO && retBand == 1) {
+                char* pEnd = &ubandStr[0];
+                ubandUint64 = strtoull(ubandStr, &pEnd, 10);
+                // Only update if Twilio Super SIM and not set to correct bands
+                if (pEnd - ubandStr > 0 && ubandUint64 != 6170) {
+                    // Enable Cat-M1 bands 2,4,5,12 (AT&T), 13 (VZW) = 6170
+                    parser_.execCommand(UBLOX_UBANDMASK_TIMEOUT, "AT+UBANDMASK=0,6170");
+                    // Not checking for error since we will reset either way
+                    reset = true;
+                    disableLowPowerModes = false;
+                }
+            }
+        }
+        if (reset) {
+            CHECK_PARSER_OK(parser_.execCommand("AT+CFUN=15,0"));
+            HAL_Delay_Milliseconds(2000);
+
+            CHECK(waitAtResponseFromPowerOn(state));
+
+            // Checking for SIM readiness ensures that other related commands
+            // like IFC or PSM/eDRX won't error out
+            CHECK(checkSimReadiness());
+            // Prevent modem from immediately dropping into PSM/eDRX modes
+            // which (on 05.12) may be enabled as soon as the UMNOPROF has taken effect
+            if (disableLowPowerModes) {
```

Comment:
@avtolstoy Could you also please shed some light on this?
If for some reason, UMNOPROF is set correctly with the first attempt, it will not try to set UMNOPROF again. But during the first time's reset, if any of the `checkSimReadiness` or `waitAtResponseFromPowerOn` have failed, we won't disable psm/edrx and since the CHECK() fails, we exit with "Failed to perform earlier init..". Next, as a part of init, we will check UMNOPROF, but it's already set to the desired value so it won't enter `if (reset)`, in which case disablePsmEdrx() will not be called. It's a very corner case, but just wanted to put a thought out there.

So here's a sequence for example (may be Twilio):
```
1. UMNOPROF reads as 0
2. UMNOPROF set to 100
3. Reset using CFUN=15.
4. Enters `if (reset)`. Let's say `waitAtResponseFromPowerOn` or `checkSimReadiness` has failed for some reason. We go no further and exit
5. Next round, we check UMNOPROF, it's already 100, so we don't enter the `if (reset)` loop which has `disablePsmEdrx` , hence potentially not disabling psm/edrx. 
```
This may be a corner case that I don't know if it can happen.

Reply:
@keeramis It will still happen [here](https://github.com/particle-iot/device-os/pull/2317/files#diff-a8e99f7ec884e049b049347a881b289b8edd5641af45e8d407cb97a78fdf7550R1368), right?

Reply:
Yes, that is correct. It cannot escape from disablement - that is why, kept those calls [there](https://github.com/particle-iot/device-os/pull/2317/files#diff-a8e99f7ec884e049b049347a881b289b8edd5641af45e8d407cb97a78fdf7550R1368) intact. Cool. That should be OK.